### ISSUE TITLE
ogn-rf and gsm_scan require deprecated libjpeg.so.8 (v0.2.5)

### ISSUE THREAD

**Issue by kerel-fs** [glidernet/ogn-rf#20]
Currently I'm trying to package rtlsdr-ogn 0.2.5 for Debian (as a binary package, since `ogn-decode` is closed-source). Unfortunately `ogn-rf` and `gsm_scan` were built against libjpeg8, which isn't available in debian jessie and newer.
Please build against libjpeg62-turbo in the next release of rtlsdr-ogn to fix this issue.

**Comment by Romeo-Golf** [comment]
from memory it was libjpeg-dev which I found to be required in place of libjpeg8 ..

**Comment by kerel-fs** [comment]
@Romeo-Golf Yes, thanks.

`libjpeg-dev` is a dummy/virtual package which defaults to the default Debian implementation of libjpeg.so.62 JPEG library (on debian jessie this means `libjpeg62-turbo`).
That said, rtlsdr-ogn should be built against `libjpeg.so.62`, which is provided by `libjpeg-dev`.

**Comment by snip** [comment]
libjpeg-dev is for compilation time not running time isn't it?

**Comment by kerel-fs** [comment]
For v0.2.5 during runtime the shared library `libjpeg.so.8` is required, provided by `libjpeg8`.
During build time the headers were provided by `libjpeg8-dev`.

For the next release, the shared library `libjpeg.so.62` will be required, provided by `libjpeg62` aka. `libjpeg62-turbo`,
if you use the headers provided by `libjpeg-dev` aka. `libjpeg62-turbo-dev`.

**Comment by kerel-fs** [comment]
PS. Anyway, I already solved this ""problem"" by building a source package for ogn-rf. :)
I will publish it when I managed to install the configuration file and the named pipe (fifo) properly.

**Comment by snip** [comment]
I think the fifo can be created by ogn-rf itself instead of externally creating it.

**Comment by pyrog** [comment]
> I think the fifo can be created by ogn-rf itself instead of externally creating it.

👍

**Comment by svoop** [comment]
@kerel-fs 

Same problem here: Another piece of software requires libjpeg-turbo and is therefore blocked by v0.2.5. The OS is Gentoo, building against `libjpeg.so.62` in v0.2.6 will solve this.

Will you be cutting v0.2.6 any time soon? I don't want to jangle your nerves, I'm just asking because I'd like to move the receiver to it's proper location on top of the hangar – and in case anything goes wrong upping rtlsdr-ogn, it'll be much harder to plug onto the UART socket up there 😁

**Comment by svoop** [comment]
@kerel-fs I've just upped the ebuild to 0.2.6 and switched to libjpeg-turbo as [suggested by you](https://github.com/glidernet/ogn-rf/issues/20#issuecomment-279256958). However, it bombs: `error while loading shared libraries: libjpeg.so.8`. Wasn't the new release supposed to build against `libjpeg.so.62`?

**Comment by kerel-fs** [comment]
Yes, it was supposed to build against `libjpeg.so.62`, but apparently the update to 0.2.6 had to be done in a hurry so this issue wasn't considered. I guess it will be considered in a future update.

**Comment by hb9eue** [comment]
After upgrading from Raspian 10 to 11 libjpeg8 ist NOT available anymore.
ogn-rf and ogn-decode depend from it and do not run anymore on my raspberry.

**Comment by fe00h** [comment]
Note that you can recompile the latest `ogn-rf` sources to avoid it depending on libjpeg8

**Comment by DooMMasteR** [comment]
ogn-decode still needs the libjpeg8 though...

**Comment by VladimirVLF** [comment]
You can build it from sources. New Raspbian OS have all build tools pre-installed, so just run:

```bash
wget -qO- http://www.ijg.org/files/jpegsrc.v8d.tar.gz | tar -xz
cd jpeg-8d/
./configure --libdir=/usr/lib/
make && sudo make install
rm -fr jpeg-8d/
```
